docs(animations): improve the aio animations introduction (#42885)
angular/angular: aio/content/guide/complex-animation-sequences.md

So far, we've learned simple animations of single HTML elements. Angular also lets you animate coordinated sequences, such as an entire grid or list of elements as they enter and leave a page. You can choose to run multiple animations in parallel, or run discrete animations sequentially, one following another.

The functions that control complex animation sequences are:

* `query()` finds one or more inner HTML elements.
* `stagger()` applies a cascading delay to animations for multiple elements.

> * <code><a href="api/animations/group" class="code-anchor">group</a>()</code> runs multiple animation steps in parallel.

Author: here and in other places I make this change, it makes for an uglier markdown but a much nicer and more consistent outcome

before:
![Peek 2021-07-17 18-17](https://user-images.githubusercontent.com/61631103/126044847-928dd5c4-d484-492b-8d86-e11b89263356.gif)

after:
![Peek 2021-07-17 18-18](https://user-images.githubusercontent.com/61631103/126044852-22c40f78-1990-47e7-a111-46f81742efc5.gif)

The `stagger()` function allows you to define a timing gap between each queried item that is animated and thus animates elements with a delay between them.

The following example demonstrates how to use the `query()` and `stagger()` functions to animate a list (of heroes) adding each in sequence, with a slight delay, from top to bottom.

* Use `query()` to look for an element entering the page that meets certain criteria.

> * For each of these elements, use `style()` to set the same initial style for the element. Make it transparent and use `transform` to move it out of position so that it can slide into place.

Author: not sure about the word `invisible` makes me think of the css visibility property...

I think "transparent" is actually a better word here (it is the synonym of opaque)

## Parallel animation using group() function

> You've seen how to add a delay between each successive animation. But you may also want to configure animations that happen in parallel. For example, you may want to animate two CSS properties of the same element but use a different `easing` function for each one. For this, you can use the animation <code><a href="api/animations/group" class="code-anchor">group</a>()</code> function.

Reviewer: ```suggestion
Previous sections describe how to add a delay between successive animations. But you might also want to configure animations that happen in parallel. For example, you might want to animate two CSS properties of the same element but use a different `easing` function for each one. To accomplish this task, you can use the animation <code><a href="api/animations/group" class="code-anchor">group</a>()</code> function.
```

<div class="alert is-helpful">

**Note:** The <code><a href="api/animations/group" class="code-anchor">group</a>()</code> function is used to group animation *steps*, rather than animated elements.
</div>

The following example, uses <code><a href="api/animations/group" class="code-anchor">group</a></code>s on both `:enter` and `:leave` for two different timing configurations, thus applying two independent animations to the same element in parallel.
<!-- I find this example a bit confusing, I think you be better to have a simpler example with just one group instead of

> making things more complex -->

Author: sorry, regardless on my typo I put this here for myself as a reminder I will remove it

but yeah I do believe that the example here is unnecessarily complex and that it could be simpler

> making things more complex -->

Author: sorry! gosh I can't believe this wasn't already removed :sweat: :bow: :pray: +

It's been ages since I created this PR, not even sure why I found this too complicated, we can ignore this for now I think :slightly_smiling_face:

## Filter animation example

> Let's take a look at another animation example. We have a search input which filters some heroes.

Author: I have no idea what live example page this is mentioning (I mean, I've found it in the code, but I could not find it in the docs)

Reviewer: Hm. Are we missing a link? If so, let's add it! Can you tell me where the example is?

Author: Sorry, it's been a couple of weeks since I made this change so I am not 100% sure, but I think that the issue is that here it mentions this example but it's not like it's introducing it and I could not find anywhere were it was actually uses, so I think that my understanding was that probably such an example existed but it was removed or something but still referenced here

but the code is actually still there, so it may actually be used somewhere....

so yeah I am not completely sure :sweat_smile: , I'll need to have another look at it and give you more details, sorry :pray: :bow:

Author: ok.... so sorry I was mistaken! :bow: 

Here's the example: https://angular.io/guide/example-apps-list#animations

The thing is that that is the only single place in the whole site which actually links to the example....

I don't think we should just here say something and link the page I just mentioned, I think we should mention the live example somewhere (like at the start of the guides) and then refer to that :thinking: 

What do you think @aikidave do you agree?

Author: @aikidave I am so sorry :bow: :bow: :bow: :bow: :bow: :bow: :bow: :bow: :bow: :bow: 

The guide already presents the example:
![Screenshot at 2021-09-01 23-09-59](https://user-images.githubusercontent.com/61631103/131752045-16f998ee-9bcf-47e5-9aaa-b45580e972c4.png)

(I missed that because if just has it as `<live-example></live-example>` I didn't know that was another way to include it :sweat: )

So should I just revert the sentence to its original form? :thinking: 

> ps: I admit that I may just have been stupid... but if I got confused/lost about this I believe it is possible that it is not really crystal clear... I think my issue was that this is just quickly mentioned like this at the beginning of the guide, and then mentioned again only a couple of times many sections/guides later... I wonder if we can do anything about that :sweat:

Author: Sorry I am not sure if your suggestion has a typo of something (I don't get the `in the .` part :sweat:)

___

Anyways my issue here is a general one, seems like we have this live example and I actually I think that all (or most of) the examples presented in this guide are from the live example.

If you're ok with it I am thinking to just revert it back to the original text (as it is sort of right, it's just that I didn't get it) and then if we want I can go back to the guide and try to make generally clear that the examples are taken from the live/example one.

What do you say about that? :slightly_smiling_face:

Author: Actually this was involved into a merge conflict so I too the occasion to revert it back to it's original form, @aikidave please let me know if it's ok :sweat:

Reviewer: I have a plan to do some additional work on these topics; your efforts are making that work much easier, @dario-piotrowicz !

As the merge conflict caused you to revert these sentences back to their original form, I'm fine if we leave them as is. Let's not have perfect be the enemy of good!

Author: wow I am really glad if I could be any useful!!!! :smile: 

Yeah the merge conflict was nothing breaking really, but since I was already thinking of reverting it back I just took it as a good chance to just do that :sweat_smile: 

Any yeah I agree with the perfection thing, this PR's been open for ages, the priority should be hopefully to get it merged (as it is improving things anyways) and take care of other stuff later on :slightly_smiling_face: 

> PS: if there'll be anything I could do to assist you please let me know! :smile:

Let's take a look at another animation example. We have a search input which filters some heroes.

The filter works in real time as you type. Elements leave the page as you type each new letter and the filter gets progressively stricter. The heroes list gradually re-enters the page as you delete each letter in the filter box.

The HTML template contains a trigger called `filterAnimation`.

<code-example path="animations/src/app/hero-list-page.component.html" header="src/app/hero-list-page.component.html" region="filter-animations"></code-example>

The component file contains three transitions.
The `filterAnimation` in the component's decorator contains three transitions.

<code-example path="animations/src/app/hero-list-page.component.ts" header="src/app/hero-list-page.component.ts" region="filter-animations" language="typescript"></code-example>

> The animation does the following:

Author: This was not the best wording I believe, for example we had something like:

- The animation does the following:
 * Ignores any animations ...
 
doesn't really make too much sense, does it? :confused:

Reviewer: What about:

"The code in this example performs the following tasks:" ?

Author: yeah it makes much more sense than what I put there :sweat_smile:

> * Doesn't animate any element when the user first opens or navigates to this page (the filter animation narrows what is already there, so it only works on elements already exist in the DOM).

Reviewer: ```suggestion
* Skips animating elements when the user first opens or navigates to this page (the filter animation narrows what is already there, so it only works on elements that already exist in the DOM).
```

What do you think of this wording?

<code-example path="animations/src/app/hero-list-page.component.html" header="src/app/hero-list-page.component.html" region="filter-animations"></code-example>

The `filterAnimation` in the component's decorator contains three transitions.

<code-example path="animations/src/app/hero-list-page.component.ts" header="src/app/hero-list-page.component.ts" region="filter-animations" language="typescript"></code-example>

The code in this example performs the following tasks:

> * Skips animating elements when the user first opens or navigates to this page (the filter animation narrows what is already there, so it only works on elements that already exist in the DOM).

Reviewer 2: "Skips animating elements" is a bit confusing here (at first I was interpreting it as "skips elements that are animated").
How about:

```suggestion
* Skips animations when the user first opens or navigates to this page (the filter animation narrows what is already there, so it only works on elements that already exist in the DOM).
```

Author: ah.... yeah I understand what you mean :+1: 

(like "animating" could interpreted as the verb for the execution or the adjective for the elements)
